### PR TITLE
Update deployment guide to support KVM

### PR DESCRIPTION
**Pull request by cmurphy**
The existing bare metal deployment guide is also sufficient for
deployment on KVM machines. This change makes minor modifications to
make that support explicit, adds a note about how to manage virtual
machines, and fixes the load balancer instructions to be clear that the
guide on setting up a load balancer is applicable to both bare metal and
KVM hosts.

Fixes https://github.com/SUSE/avant-garde/issues/1165

### CODE REVIEW

**Comment by r0ckarong** [comment]
@kkaempf Does this mean we are officially adding KVM support? If not I don't really know what to do with this.

**Comment by cmurphy** [comment]
@r0ckarong I believe the goal is to officially support KVM according to https://github.com/SUSE/avant-garde/issues/996

It may be a good idea to hold off merging this until the CI is in place.

**Comment by kkaempf** [comment]
> @kkaempf Does this mean we are officially adding KVM support? If not I don't really know what to do with this.
Yes: https://github.com/SUSE/avant-garde/issues/996

**Comment by kkaempf** [comment]
> @r0ckarong I believe the goal is to officially support KVM according to [SUSE/avant-garde#996](https://github.com/SUSE/avant-garde/issues/996)
> 
> It may be a good idea to hold off merging this until the CI is in place.

It can be merged (and announced) as soon as engineering declares "we're ready to support it". CI support is a reasonable prerequisite, engineering can raise more prerequisites if needed.

**Comment by Klaven** [comment]
this should be held till CI is supported. we are working on this. can be tracked here https://github.com/SUSE/avant-garde/issues/1227

File: adoc/deployment-bare-metal-or-kvm.adoc [hunk 1]
```diff
@@ -1,19 +1,23 @@
...
+== Deployment on Bare Metal or KVM
 
 .Preparation Required
 [NOTE]
 You must have completed <<deployment.preparations>> to proceed.
 
+[NOTE]
+====
+If deploying on KVM virtual machines, you may use a tool such as `virt-manager`
+to configure the virtual machines and begin the {sle} 15 SP1 installation.
```
Comment (kkaempf) [inline adoc/deployment-bare-metal-or-kvm.adoc]: '15 SP1' should also be a macro in order to prepare for SP2 later this year.

File: adoc/deployment-bare-metal-or-kvm.adoc [hunk 2]
```diff
@@ -30,7 +34,7 @@ you must adjust the {ay} file to your needs according to the requirements.
 Refer to the official {ay} documentation for more information: link:https://documentation.suse.com/sles/15-SP1/single-html/SLES-autoyast/#book-autoyast[{ay} Guide].
```
Comment (kkaempf) [inline adoc/deployment-bare-metal-or-kvm.adoc]: same here

**Comment by kkaempf** [comment]
approved

**Comment by jodavis-suse** [comment]
LGTM too

**Comment by dirkmueller** [comment]
@cmurphy rebase?

**Comment by cmurphy** [comment]
@dirkmueller done

But I think this is still waiting on CI to be in place before we can declare it officially supported?

**Comment by dannysauer** [comment]
I thought CI was in place; we were just waiting for Terraform 1.12.  Which is in 4.1.2, whenever we officially give that release the blessing.

**Comment by r0ckarong** [comment]
@jordimassaguerpla What is the status of this? Will we ship this with 4.1.2?

**Comment by jordimassaguerpla** [comment]
> @jordimassaguerpla What is the status of this? Will we ship this with 4.1.2?

Yes it is included. QA asked us to create a maint. incident, so this means they are fine with https://github.com/SUSE/avant-garde/projects/16

**Pull request merged**